openeo/extra/job_management/__init__.py

```diff
@@ -540,6 +540,16 @@ def _job_update_loop(
                         stats["job_db persist"] += 1
                         total_added += 1
 
+        # Act on jobs
+        for job, row in jobs_error:
+            self.on_job_error(job, row)
+
+        for job, row in jobs_cancel:
+            self.on_job_cancel(job, row)
+
+        for job, row in jobs_done:
+            self.on_job_done(job, row)
```

Comment:
I would do the `on_job_done` before the `on_job_error`. It's probably more important to collect finished results than to collect the failures